packages/camera/camera/ios/Classes/FLTCam.m

```diff
@@ -220,11 +221,30 @@ - (void)captureToFile:(FLTThreadSafeFlutterResult *)result API_AVAILABLE(ios(10)
     return;
   }
 
-  [_capturePhotoOutput
-      capturePhotoWithSettings:settings
-                      delegate:[[FLTSavePhotoDelegate alloc] initWithPath:path
-                                                                   result:result
-                                                                  ioQueue:self.photoIOQueue]];
+  FLTSavePhotoDelegate *savePhotoDelegate = [[FLTSavePhotoDelegate alloc]
+           initWithPath:path
+                ioQueue:self.photoIOQueue
+      completionHandler:^(NSString *_Nullable path, NSError *_Nullable error) {
+        dispatch_async(self.captureSessionQueue, ^{
+          // Dispatch back to capture session queue to delete reference.
+          // Retain cycle is broken after the dictionary entry is cleared.
+          // This is to keep the behavior with the previous `selfReference` approach in the
+          // FLTSavePhotoDelegate, where delegate is released only after capture completion.
+          self.inProgressSavePhotoDelegates[@(settings.uniqueID)] = nil;
```

Comment:
Or this, now that you're not mocking the dictionary subscripting. Not important though.
```suggestion
          [self.inProgressSavePhotoDelegates removeObjectForKey:@(settings.uniqueID)];
```